packages/integrations/node/src/middleware.ts

```diff
@@ -51,8 +51,12 @@ async function writeWebResponse(app: NodeApp, res: ServerResponse, webResponse:
 
 	res.writeHead(status, Object.fromEntries(headers.entries()));
 	if (webResponse.body) {
-		for await (const chunk of responseIterator(webResponse) as unknown as Readable) {
-			res.write(chunk);
+		try {
+			for await (const chunk of responseIterator(webResponse) as unknown as Readable) {
+				res.write(chunk);
+			}
+		} catch(err: any) {
+			res.write(err + '');
```

Comment:
Is it safe to write the error like this? In dev we usually don't care, but in prod you might not want errors to show in the user's browser

Reply:
I was thinking this is what we did if the error happened before the response, but I guess not: https://github.com/withastro/astro/blob/main/packages/astro/src/core/app/index.ts#L210

So maybe log and write out `'Internal server error'` instead?

Reply:
That's what I would personally go for, with a 500 status code

Reply:
Can't set the status code at this point, the response is already sent. So this is just a matter of what goes in the HTML. Really the application should be handling errors themselves, this is a fallback case.

Reply:
Ah yes, fair. Well, I stil think show Internal Server Error is safer, just in case